File: src/main/java/cn/zzwtsy/pu/database/InitDataBase.java
package cn.zzwtsy.pu.database;

import cn.zzwtsy.pu.PuCampus;

import java.io.File;
import java.sql.SQLException;

import static cn.zzwtsy.pu.tools.DataBaseStatic.DB_FILE_PATH;

/**
 * 初始化数据库
 *
 * @author zzwtsy
 * @since 2022/11/25
 */
public class InitDataBase {
    public void initDataBase() {
        File dataFile = new File(DB_FILE_PATH);
        if (!dataFile.exists()) {
            String creatSchoolInfoTableSql = """
                    CREATE TABLE "T_school_info" (
                    "schoolName" TEXT(30) NOT NULL,
                    "email" TEXT(20) NOT NULL,
                    "schoolId" TEXT(20) NOT NULL,
                    "displayOrder" TEXT(30) NOT NULL,
                    PRIMARY KEY ("schoolName")
                    );""";
            try {
                if (DataBaseHelper.executeUpdate(creatSchoolInfoTableSql) == -1) {
                    PuCampus.INSTANCE.getLogger().error("数据库创建失败");
                } else {
                    PuCampus.INSTANCE.getLogger().info("数据库创建成功");
                }
            } catch (SQLException e) {
                throw new RuntimeException(e);
            }
        } else {
            PuCampus.INSTANCE.getLogger().info("数据库已存在");
        }
    }
}

File: src/main/kotlin/cn/zzwtsy/pu/database/InitDatabase.kt
package cn.zzwtsy.pu.database

import cn.zzwtsy.pu.PuCampus
import cn.zzwtsy.pu.tools.*
import java.io.File
import java.sql.DriverManager
import java.sql.SQLException

/**
 * 初始化数据库
 * @author zzwtsy
 * @date 2023/03/21
 * @constructor 创建[InitDatabase]
 */
class InitDatabase {

    /**
     * 判断数据库是否存在，不存在则创建 sqlite 数据库和表
     */
    fun init() {
        if (File(DB_PATH).exists() || File("$DB_PATH/$DB_NAME").exists()) {
            return
        }
        //创建目录
        File(DB_PATH).mkdirs()
        PuCampus.INSTANCE.logger.info("数据库不存在，初始化数据库")
        try {
            Class.forName(DRIVER_NAME)
        } catch (e: ClassNotFoundException) {
            PuCampus.INSTANCE.logger.error("连接数据库失败", e)
        }
        try {
            val executeUpdateStatus = DriverManager
                .getConnection(SQL_CONNECT_URL)
                .prepareStatement(USER_TABLE_SQL)
                .executeUpdate()
            if (executeUpdateStatus == 0)
                PuCampus.INSTANCE.logger.info("创建 user 表成功")
            else
                PuCampus.INSTANCE.logger.error("创建 user 表失败")
        } catch (e: SQLException) {
            PuCampus.INSTANCE.logger.error("创建 user 表失败", e)
        }
    }
}